engine/table/src/main/java/io/deephaven/stream/StreamToTableAdapter.java

```diff
@@ -329,9 +344,12 @@ private void doRefresh() {
             rowSet.removeRange(newSize, oldSize - 1);
         }
 
-        table.notifyListeners(new TableUpdateImpl(RowSetFactory.flat(newSize),
-                RowSetFactory.flat(oldSize), RowSetFactory.empty(),
-                RowSetShiftData.EMPTY, ModifiedColumnSet.EMPTY));
+        final QueryTable localTable = tableRef.get();
+        if (localTable != null) {
+            localTable.notifyListeners(new TableUpdateImpl(RowSetFactory.flat(newSize),
+                    RowSetFactory.flat(oldSize), RowSetFactory.empty(),
+                    RowSetShiftData.EMPTY, ModifiedColumnSet.EMPTY));
+        }
```

Comment:
I'm ambivalent. We could, since we think that without the child table we have no reason to continue existing, but it's not strictly necessary. If `tableRef` has been cleared, the cleanup handler thread should drain its `RetainedReferenceTracker` from the reference queue, which should drop our reference count to 0. Probably better to be aggressive, though, in case someone has us in a `LivenessScope` keeping us active when there's no reason for it. I will make two "suggestion" comments.